graphium/config/_loader.py

```diff
@@ -589,15 +589,15 @@ def get_checkpoint_path(config: Union[omegaconf.DictConfig, Dict[str, Any]]) ->
 
     cfg_trainer = config["trainer"]
 
-    path = config.get("ckpt_name_for_testing", "last.ckpt")
-    if path in GRAPHIUM_PRETRAINED_MODELS_DICT or fs.exists(path):
-        return path
-
     if "model_checkpoint" in cfg_trainer.keys():
-        dirpath = cfg_trainer["model_checkpoint"]["dirpath"]
-        path = fs.join(dirpath, path)
+        dirpath = cfg_trainer["model_checkpoint"]["dirpath"] + str(cfg_trainer["seed"]) + "/"
```

Comment:
Can you create a unique identifier that you can associate with a specific run? Perhaps take the identifier from Wandb?

Reply:
Unless you want to add an ID that is only programmatically accessible, I would leave it to the config and not hard-code the seed in the path since it is pretty subjective where you want to save the models. Note that setting the dirpath to include the seed can currently also be achieved by changing the config - either directly or through the CLI.

Reply:
We have the issue that we want identifiable checkpoints, and non- overwriting checkpoints. How do you suggest we move forward?

Reply:
I understand the use-case! 

As I said before, I suggest using the config. I would set a sensible default, but then leave it up to the user. 

E.g. the line that is hard coded here can already be configured with [`hydra`s override syntax](https://hydra.cc/docs/advanced/override_grammar/basic/), using [OmegaConf's interpolation](https://omegaconf.readthedocs.io/en/2.3_branch/usage.html#variable-interpolation): 
```sh
graphium-train ... "trainer.model_checkpoint.dirpath='/path/to/checkpoints/${constants.seed}'"
```

In case of a sweep, you might want to use something more specific, e.g.: 
```sh
graphium-train -m ... "trainer.model_checkpoint.dirpath='/path/to/checkpoints/${now:%Y-%m-%d_%H-%M-%S}'"
```

To prevent most users from having to set a value themselves, you could include a sensible default in the config files. Current defaults (e.g. `models_checkpoints/neurips2023-small-gcn/`) are too specific and not consistent. I would e.g. save to the [current working dir](https://hydra.cc/docs/configure_hydra/workdir/) by default: 

```yaml
trainer:
  model_checkpoint:
    dirpath: ${hydra.run.dir}
```

Reply:
Thanks for the suggestion. I'm still getting used to hydra and didn't know you could put the date-time directly in the Yaml! What do you think of [this commit](https://github.com/datamol-io/graphium/pull/449/commits/da31797946fe2f094c355aca452f0df5d872fcbe)? Making it the default in all the `hydra-configs`

Reply:
Or do you mean that, by default, the `{hydra.run.dir} includes the date-time?

Reply:
I agree that we have to leave the checkpoint path configuration somewhat up to the user.

However, the seed is an important culprit leading to overwriting (or unclear identity) of model checkpoints. I post-pended it to the `dirpath `because we also used to have the following in `load_trainer`:
```
if "model_checkpoint" in cfg_trainer.keys():
    cfg_trainer["model_checkpoint"]["dirpath"] += str(cfg_trainer["seed"]) + "/"
    callbacks.append(ModelCheckpoint(**cfg_trainer["model_checkpoint"]))
```

Reply:
I am using `graphium-train ... "trainer.model_checkpoint.dirpath='/path/to/checkpoints/${constants.seed}'"` as proposed by Cas to reasonbly structure checkpoints

Reply:
Why not use the full date and time? Much better than the random seed. And avoids collision when doing sweeps with pre-selected seeds. And also interpretable.

Reply:
Yes, I agree. It was necessary at the time the branch was created because the seed was hard-coded into the `dirpath` back then... I think it is good as of now using the date :)

Reply:
Changed function discussed here to its current state in the main branch. The function was outdated in this branch because we added it to https://github.com/datamol-io/graphium/pull/441/files where it was modified to accommodate further changes :)